### PR TITLE
Adds some ESGF error checking, velo_new_folder, and velo_save_file

### PR DESCRIPTION
**Pull request by sterlingbaldwin**

### CODE REVIEW

**Comment by sterlingbaldwin** [comment]
I need to figure out whats going wrong here...

File: apps/velo/get_file.py [hunk 1]
```diff
@@ -20,4 +20,5 @@ def get_file(remote_file_path, local_path, filename, site_user, velo_username, p
         raise
 
 
-get_file(sys.argv[1], sys.argv[2], sys.argv[3], sys.argv[4], sys.argv[5], sys.argv[6])
+get_file(sys.argv[1], sys.argv[2], sys.argv[3],
+         sys.argv[4], sys.argv[5], sys.argv[6])
```
Comment (chaosphere2112) [inline apps/velo/get_file.py]: Can we stick these into variables with useful names first so we have some kind of documentation as to what all of these args mean?

File: apps/velo/new_folder.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        if velo_api.create_folder(foldername) >= 0:
+            print "Creaded new folder"
+            return 0
+        else:
+            print 'Failed to create new folder'
+    except:
+        raise
+
+
+new_folder(sys.argv[1], sys.argv[2], sys.argv[3])
```
Comment (chaosphere2112) [inline apps/velo/new_folder.py]: See comment in get_file.py

File: apps/velo/save_file.py [hunk 3]
```diff
@@ -3,32 +3,31 @@
...
         else:
+            print 'Error saving file', local_path, filename
             return -1
     except:
         raise
 
 
 save_file(sys.argv[1], sys.argv[2], sys.argv[3],
-          sys.argv[4], sys.argv[5], sys.argv[6])
+          sys.argv[4], sys.argv[5])
```
Comment (chaosphere2112) [inline apps/velo/save_file.py]: Let's wrap all of these calls with a sys.exit so the return values actually get caught

File: apps/web_fe/models.py [hunk 4]
```diff
@@ -11,6 +12,7 @@ class TileLayout(models.Model):
 
 
 class Credential(models.Model):
+    # site_user_name = models.ForeignKey(User)
```
Comment (chaosphere2112) [inline apps/web_fe/models.py]: no commented code!

File: apps/web_fe/views.py [hunk 5]
```diff
@@ -248,20 +255,23 @@ def jspanel(request):
...
+            node_list = []
+            return HttpResponse(render_template(request, "web_fe/grid.html", {'nodes': node_list}))
         f = StringIO(r.content)
         out = open('scripts/registration.xml', 'w')
         out.write(f.read())
         out.close()
     except Exception as e:
         import traceback
-        if 'requests.exceptions.ConnectTimeout' in str(sys.exc_info()):
+        if 'requests.exceptions.ConnectTimeout' in str(sys.exc_info()) or 'requests.exceptions.ConnectionError' in str(sys.exc_info()):
```
Comment (chaosphere2112) [inline apps/web_fe/views.py]:
```
from requests.exceptions import ConnectTimeout, ConnectionError
if type(e) in (ConnectTimeout, ConnectionError):
```

File: apps/web_fe/views.py [hunk 6]
```diff
@@ -364,6 +373,8 @@ def node_info(request):
     if request.method == 'POST':
         try:
             from xml.etree.ElementTree import parse
+            import os.path
```
Comment (chaosphere2112) [inline apps/web_fe/views.py]: Debugging code :bomb:

File: apps/web_fe/views.py [hunk 7]
```diff
@@ -565,18 +576,52 @@ def velo_save_file(request):
     if request.method == 'POST':
         try:
             incomming_file = json.loads(request.body)
-            remote_path = incomming_file['remote_path']
+            # remote_path = incomming_file['remote_path']
```
Comment (chaosphere2112) [inline apps/web_fe/views.py]: Commented code :bomb:

Comment (chaosphere2112) [inline apps/web_fe/views.py]: also incoming*

File: apps/web_fe/views.py [hunk 8]
```diff
@@ -565,18 +576,52 @@ def velo_save_file(request):
...
+                ['python', './apps/velo/save_file.py', text, filename, cred.site_user_name, cred.service_user_name, cred.password], stdout=PIPE)
+            (out, err) = process.communicate()
+            exit_code = process.wait()
+            if exit_code >= 0 and 'File saved' in out:
+                return HttpResponse(status=200)
+            else:
+                print out, err
+                return HttpResponse(status=500)
+        except Exception as e:
+            import traceback
```
Comment (chaosphere2112) [inline apps/web_fe/views.py]: If you really want all of this debugging logging stuff, put it in a "debug_exception" function and save yourself some copy-pasta.

File: apps/web_fe/views.py [hunk 9]
```diff
@@ -565,18 +576,52 @@ def velo_save_file(request):
...
+
             process = Popen(
-                ['python', './apps/velo/save_file.py', text, local_path, remote_path, site_user, velo_user, velo_pass, filename], stdout=PIPE)
+                ['python', './apps/velo/new_folder.py', foldername, cred.service_user_name, cred.password], stdout=PIPE)
             (out, err) = process.communicate()
             exit_code = process.wait()
-            out = out.splitlines(True)[1:]
-            if exit_code == 0:
+
+            if exit_code == 0 and 'Creaded new folder' in out:
```
Comment (chaosphere2112) [inline apps/web_fe/views.py]:
```
created*
```

File: static/js/tile.js [hunk 10]
```diff
@@ -233,60 +245,73 @@ $(document).ready(function() {
 											'<label for="velo_username" class="sr-only">User name:</label>',
 											'<input type="text" id="velo_username" name="velo_username" class="form-control" placeholder="User Name">',
 											'<label for="velo_password" class="sr-only">Password:</label>',
-											'<input type="text" id="velo_password" name="velo_password" class="form-control" placeholder="Password">',
+											'<input type="password" id="velo_password" name="velo_password" class="form-control" placeholder="Password">',
```
Comment (chaosphere2112) [inline static/js/tile.js:248]: :+1:

File: templates/base.html [hunk 11]
```diff
@@ -13,6 +13,7 @@
     <script src="https://code.jquery.com/jquery-2.1.4.min.js"></script>
     <script src="https://code.jquery.com/ui/1.11.4/jquery-ui.min.js"></script>
     <link rel="stylesheet" href="https://code.jquery.com/ui/1.11.4/themes/smoothness/jquery-ui.css">
+    <link rel="shortcut icon" href="/static/media/favicon.ico">
```
Comment (chaosphere2112) [inline templates/base.html]:
```
{{static}}
```

File: templates/web_fe/grid.html [hunk 12]
```diff
@@ -5,14 +5,14 @@
...
-    <link rel="stylesheet" type="text/css" href="/static/css/MetroJs.css">
+    <link rel="stylesheet" type="text/css" href="{{static}}/css/MetroJs.css">
     <script type="text/javascript" src="/static/js/MetroJs.js"></script>
 
   <!-- Custom Grid -->
-    <link rel="stylesheet" type="text/css" href="static/css/dashboard-style.css">
-    <script type="text/javascript" src="static/js/grid_layouts.js"></script>
-    <script type="text/javascript" src="static/js/tile.js"></script>
-    <link rel="stylesheet" type="text/css" href="static/css/tile.css">
+    <link rel="stylesheet" type="text/css" href="/static/css/dashboard-style.css">
```
Comment (chaosphere2112) [inline templates/web_fe/grid.html]: `{{static}}`

File: templates/base.html (hunk 11 above)
Comment (chaosphere2112) [inline templates/base.html]: Missed one :smile:

File: templates/web_fe/grid.html [hunk 13]
```diff
@@ -21,19 +21,19 @@
...
+  <link rel="stylesheet" href="{% static "css/twilight.css" %}">
+  <link rel="stylesheet" href="{% static "css/3024-day.css" %}">
+  <link rel="stylesheet" href="{% static "css/codemirror.css" %}">
 
   <!-- mTree -->
-  <link rel="stylesheet" href="static/css/mtree.css">
+  <link rel="stylesheet" href="{% static "css/mtree.css" %}">
   <script src='http://cdnjs.cloudflare.com/ajax/libs/velocity/0.2.1/jquery.velocity.min.js'></script> 
-  
+  <!-- {{ STATIC_URL }} {% static "css/dashboard-style.css" %} -->
```
Comment (chaosphere2112) [inline templates/web_fe/grid.html]: You should remove this comment

Comment (sterlingbaldwin) [inline templates/web_fe/grid.html]: Ooops!

**Pull request merged**